src/frontend/src/binder/expr/function.rs

```diff
@@ -77,12 +80,38 @@ impl Binder {
                     }
                     None => Condition::true_cond(),
                 };
+                // TODO(yuchao): handle DISTINCT and ORDER BY appear at the same time
+                if f.distinct && !f.order_by.is_empty() {
+                    return Err(ErrorCode::InvalidInputSyntax(
+                        "DISTINCT and ORDER BY are not supported to appear at the same time now"
+                            .to_string(),
+                    )
+                    .into());
+                }
+                let order_by = AggOrderBy::new(
+                    f.order_by
+                        .into_iter()
+                        .map(|e| -> Result<AggOrderByExpr> {
+                            Ok(AggOrderByExpr {
+                                expr: self.bind_expr(e.expr)?,
+                                direction: e.asc.map_or(Direction::Any, |asc| {
+                                    if asc {
+                                        Direction::Asc
+                                    } else {
+                                        Direction::Desc
+                                    }
+                                }),
```

Comment:
According to [pg doc](https://www.postgresql.org/docs/current/queries-order.html):

> The NULLS FIRST and NULLS LAST options can be used to determine whether nulls appear before or after non-null values in the sort ordering. By default, null values sort as if larger than any non-null value; that is, NULLS FIRST is the default for DESC order, and NULLS LAST otherwise.

When `NULLS FIRST|LAST` is not specified, the default is dynamically decided according to ASC/DESC. If use `bool` without `Option`, maybe it's better to rename the field to `nulls_largest`, should we do that?

Reply:
I am not sure which one is easier to use in all existing users of `Direction`: `nulls_first: bool` or `nulls_largest: bool`. Just want to show both names are valid after removing `Option`:
|asc|Option&lt;bool&gt;|nulls_first|nulls_largest|
|-|-|-|-|
|asc or default|None|false|true|
|asc or default|Some(true)|true|false|
|asc or default|Some(false)|false|true|
|desc|None|true|true|
|desc|Some(true)|true|true|
|desc|Some(false)|false|false|

Reply:
I changed the field to `bool`, and assigned it value according to the direction field. Is that OK?

https://github.com/singularity-data/risingwave/blob/22359adb329a34ac87a464555d39fd61312ff84a/src/frontend/src/binder/expr/function.rs#L94-L112

Also plan tests are fixed.